src/main/java/edu/ie3/datamodel/models/input/NodeInput.java

```diff
@@ -121,15 +122,17 @@ public NodeInputCopyBuilder copy() {
     return new NodeInputCopyBuilder(this);
   }
 
-  @Override
   public boolean equals(Object o) {
     if (this == o) return true;
     if (o == null || getClass() != o.getClass()) return false;
     if (!super.equals(o)) return false;
     NodeInput nodeInput = (NodeInput) o;
+    if (!QuantityUtil.quantityIsEmpty(vTarget)) {
+      if (QuantityUtil.quantityIsEmpty(nodeInput.vTarget)) return false;
+      if (!vTarget.isEquivalentTo(nodeInput.vTarget)) return false;
+    } else if (!QuantityUtil.quantityIsEmpty(nodeInput.vTarget)) return false;
```

Comment:
I would like you to refactor this piece of code, because it has multiple implications:

Please move this code to `QuantityUtils` in PowerSystemUtils and give it a meaningful name (e.g. `isTheSameConsideringEmpty` or `deprecatedEqualsConsideringEmpty`). There, your can also write tests for this method. This would also lower the number of untested lines, here. Please use the javadoc there, to unambiguously state, why the implementation is in the way it is. For me, it is highly unclear - or better to say misleading - what `isEquivalentTo` actually does (to me equivalence is 1 km == 1,000 m). Furthermore all those `if`-branches impose a high cognitive complexity, imposing code smells as well.

Sorry for changing out minds, but it harms more, than the solution we have had before. We just didn't saw that back then.

Please replace the single constructs here and in other classes with the aforementioned implementation.

Reply:
To clarify: isEquivalentTo works exactly like you thought, (1 km).isEquivalentTo(1000m) == true. And also, more importantly considering the issues we had in the past, (1 km).isEquivalentTo(1.0 km).
So, just to have it clear at this point, should the equals methods of our DataModels return true or false in your example case? And if false, should they return true in an Int/Double case?